From e9a29e7a69f1d50cc2393e9bd0f16b2ad58b0a9a Mon Sep 17 00:00:00 2001
From: Anthoine Bourgeois <anthoine.bourgeois@vates.tech>
Date: Thu, 13 Feb 2025 14:52:10 +0100
Subject: [PATCH] tapdisk: set generic TAPDISK_MESSAGE_MAX limit inside the
 enum

This avoid to forget to update the TAPDISK_MESSAGE_MAX define.

Signed-off-by: Anthoine Bourgeois <anthoine.bourgeois@vates.tech>
---
 drivers/tapdisk-control.c | 2 +-
 include/tapdisk-message.h | 3 +--
 2 files changed, 2 insertions(+), 3 deletions(-)

diff --git a/drivers/tapdisk-control.c b/drivers/tapdisk-control.c
index 506b203d..91ffe097 100644
--- a/drivers/tapdisk-control.c
+++ b/drivers/tapdisk-control.c
@@ -1382,7 +1382,7 @@ tapdisk_control_receive_request(struct tapdisk_ctl_conn *conn)
 	if (err)
 		goto invalid;
 
-	if (conn->request.type > TAPDISK_MESSAGE_EXIT)
+	if (conn->request.type >= TAPDISK_MESSAGE_MAX)
 		goto invalid;
 
 	conn->info = &message_infos[conn->request.type];
diff --git a/include/tapdisk-message.h b/include/tapdisk-message.h
index dceddbbe..0f73e0c9 100644
--- a/include/tapdisk-message.h
+++ b/include/tapdisk-message.h
@@ -233,10 +233,9 @@ enum tapdisk_message_id {
 	TAPDISK_MESSAGE_DISK_INFO,
 	TAPDISK_MESSAGE_DISK_INFO_RSP,
 	TAPDISK_MESSAGE_EXIT,
+	TAPDISK_MESSAGE_MAX /* This value must be the last. */
 };
 
-#define TAPDISK_MESSAGE_MAX TAPDISK_MESSAGE_EXIT
-
 static inline char *
 tapdisk_message_name(enum tapdisk_message_id id)
 {
